### ISSUE TITLE
Unable to import multiarray (migrated from Trac #33)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#595]
Original ticket http://projects.scipy.org/numpy/ticket/33
Reported 2006-03-11 by unknown, assigned to unknown.

Using numpy 0.9.5
First I tried to import numeric, this gave a  process crash.

Then I tried the following, which again crashed.

C:\Python24\Lib\site-packages\numpy\core>python
Python 2.4.2 (#67, Sep 28 2005, 12:41:11) [MSC v.1310 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.

> > > import multiarray

C:\Python24\Lib\site-packages\numpy\core>

I could copy the Windows XP response.

Colin W.

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2006-03-11

This is a known issue. It is unclear how to resolve it.  Do not import multiarray from the installed directory like this.

**Comment by thouis** [comment]
Comment in Trac by atmention:teoliphant, 2006-03-12

Flag to disable re-running of the initialization code changes this to an error message instead of a segfault.   It is still not supported to import multiarray from the core directory like this (it is not ever recommended to import multiarray directly anyway as it is an internal package implementation detail and should not be relied on). 

I tracked the segfault to the format_re = re.compile(...) line in _internal.py where the segfault was due to the string type tp_as_number table being replaced with the generic_scalar tp_as_number table.  I don't see how the code could make this happen.
